mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -1061,6 +1066,67 @@ class ConstructorPattern final
   }
 };
 
+//===----------------------------------------------------------------------===//
+// AccessorGetPointerPattern - Convert `sycl.accessor.get_pointer` to LLVM.
+//===----------------------------------------------------------------------===//
+
+class AccessorGetPointerPattern
+    : public ConvertOpToLLVMPattern<SYCLAccessorGetPointerOp>,
+      public GetMemberPattern<AccessorGetPtr>,
+      public GetMemberPattern<AccessorGetID, IDGetDim>,
+      public GetMemberPattern<AccessorGetMemRange, RangeGetDim> {
+public:
+  using ConvertOpToLLVMPattern<
+      SYCLAccessorGetPointerOp>::ConvertOpToLLVMPattern;
+
+private:
+  template <typename... Args> Value getID(Args &&...args) const {
+    return GetMemberPattern<AccessorGetID, IDGetDim>::loadValue(
+        std::forward<Args>(args)...);
+  }
+  template <typename... Args> Value getMemRange(Args &&...args) const {
+    return GetMemberPattern<AccessorGetMemRange, RangeGetDim>::loadValue(
+        std::forward<Args>(args)...);
+  }
+
+  Value getTotalOffset(OpBuilder &builder, Location loc, AccessorType accTy,
+                       OpAdaptor opAdaptor) const {
+    const auto acc = opAdaptor.getAcc();
+    const auto resTy = builder.getI64Type();
```

Comment:
Should this be:
```
const auto resTy = builder.getIndexType();
```
to allow the offset to be an i32 in 32-bit mode ?

Reply:
> error: 'llvm.getelementptr' op operand #1 must be integer or LLVM dialect-compatible vector of integer, but got 'index'

Cannot use index type for getelementptr.

Reply:
Let's look into wonder we can use the proper pointer size generally in a different PR.

Reply:
Ok. I see that clang in 32-bit mode generates the index on a GEP using i32 and it uses i64 in 64-bit mode:
32-bit mode:
```
%7 = getelementptr inbounds i32, i32* %5, i32 %6
```
64-bit mode
```
 %7 = getelementptr inbounds i32, i32* %5, i64 %6
```
We should do the same eventually but is fine to address this minor concern in a different PR.